src/main/java/com/artipie/rpm/pkg/Repodata.java

```diff
@@ -0,0 +1,106 @@
+/*
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2020 artipie.com
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included
+ * in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package com.artipie.rpm.pkg;
+
+import com.artipie.rpm.NamingPolicy;
+import com.artipie.rpm.meta.XmlPackage;
+import com.artipie.rpm.meta.XmlRepomd;
+import java.io.IOException;
+import java.nio.file.Path;
+
+/**
+ * Repodata creates repomd and files for resulting repository metadata files.
+ * @since 0.11
+ */
+public interface Repodata {
+
+    /**
+     * Creates {@link XmlRepomd} instance.
+     * @return Xml repomd
+     * @throws IOException On error
+     */
+    XmlRepomd createRepomd() throws IOException;
+
+    /**
+     * Temp dir to store metadata files.
+     * @return Path
+     */
+    Path temp();
+
+    /**
+     * Path to save resulting metadata.
+     * @param type Xml package type
+     * @param gzip Gziped metadata
+     * @return Path
+     * @throws IOException On error
+     */
+    Path resultingMetadata(XmlPackage type, Path gzip) throws IOException;
```

Comment:
@olenagerasimova this method name seems to be too  long, can't we shorten it to just `metadata`?